src/ZeekString.h

```diff
@@ -72,6 +72,7 @@ class String {
 		{ use_free_to_delete = use_it; }
 
 	const char* CheckString() const;
+	std::string ToStdString() const	{ return std::string(CheckString()); }
```

Comment:
I am not entirely sure how I feel about this one. There are a lot of cases in which strings can legitimately contain 0-bytes (which CheckString - legitimately - complains about). Which also is the reason why CheckString is often not used when accessing Zeek strings.

Maybe we could have two different accessor functions - one that creates a std::string like this - and one which just takes the full byte content and the length to construct it? Plus - documentation for both of them showing what they do.

Reply:
Wouldn't the latter just be equivalent to a call to `Len()` and `Bytes()`?

Reply:
I agree with adding comments to `CheckString()` and `ToStdString()` that clarify what exactly they do.

Reply:
In hindsight, does it make sense to add an argument to `CheckString()`/`ToStdString()` that controls whether it outputs the `<string-with-NUL>` output or just gives all of the character data up to the first null character?

Reply:
* Maybe I'd expect to just rely on implicit `std::string` conversion constructor to change `CheckString()` into `std::string` when wanting that instead of having a dedicated method

* If there were dedicated conversion methods like `ToStdString()` or `ToStringView()`, maybe I'd expect those to be more about just changing underlying storage type for convenience and use in some standard algorithm/template, and so would opt to construct those from pointer-and-length to allow for embedded null-characters, not from `CheckString()` results

* For the unit tests added here, rather than adding the `ToStdSTring()` for use in `CHECK_EQ()`, maybe the more direct solution is to either add a `zeek::String::operator==(std::string_view sv)` to compare lengths and `memcmp()` or alternatively using string literal `s` suffix: https://en.cppreference.com/w/cpp/string/basic_string/operator%22%22s

Reply:
I liked that last one and went with it. It'd be a useful method to have around anyways. I had to convert the couple of tests that check for nulls over to use `CheckString()`, but the rest remained mostly the same.